internal/publish.go

```diff
@@ -256,13 +263,42 @@ func CreateApp(ctx context.Context, config map[string]interface{}, target string
 	if err != nil {
 		return "", err
 	}
+
+	man, ok := manifest.(*ocischema.DeserializedManifest)
+	if !ok {
+		return "", fmt.Errorf("invalid manifest type, expected *ocischema.DeserializedManifest, got: %T", manifest)
+	}
+
+	b, err := man.MarshalJSON()
+	if err != nil {
+		return "", err
+	}
+
+	manMap := make(map[string]interface{})
+	err = json.Unmarshal(b, &manMap)
+	if err != nil {
+		return "", err
+	}
+
+	manMap["manifests"] = layerManifests
+
+	b1, err := json.MarshalIndent(manMap, "", "   ")
+	if err != nil {
+		return "", err
+	}
+
+	err = man.UnmarshalJSON(b1)
+	if err != nil {
+		return "", err
+	}
```

Comment:
>So this works because the manifest keeps a raw copy of the original data?
yes.

> so this is just our trick for handling this?
it's only verified  by the compose client component(s), it's never verified/checked by Registry itself.

Sure, I'll add comment to it.

I thought about another approach more logical or if you will compliant to the spec. The initial/top level App manifest could be a manifest list/index pointing to App's manifests for different architecture, then each App manifest could include the overall layer list in the legal "layers" attribute along with an App's archive. Also, it would provide opportunity to define App's compose project per architecture if it happens that there are some difference.
But, this approach will require too many changes, not sure if it's worth doing at the moment.

Reply:
yep. i think that might be more correct technically, but its still kind of (ab)using the spec and like you said - it would cause some changes which I don't think are backward compatible.